cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletUtils.java

```diff
@@ -39,4 +185,160 @@ static String getClientIpAddress(HttpServletRequest request) {
+  }
+
+  static boolean getDryRun(HttpServletRequest request) {
+    return getBooleanParam(request, DRY_RUN_PARAM, true);
+  }
+
+  static boolean throttleAddedOrRemovedBrokers(HttpServletRequest request, EndPoint endPoint) {
+    return endPoint == ADD_BROKER ? getBooleanParam(request, THROTTLE_ADDED_BROKER_PARAM, true)
+                                  : getBooleanParam(request, THROTTLE_REMOVED_BROKER_PARAM, true);
+  }
+
+  static long time(HttpServletRequest request) {
+    String timeString = request.getParameter(TIME_PARAM);
+    return (timeString == null || timeString.toUpperCase().equals("NOW")) ? System.currentTimeMillis()
+                                                                          : Long.parseLong(timeString);
+  }
+
+  static Long startMs(HttpServletRequest request) {
+    String startMsString = request.getParameter(START_MS_PARAM);
+    return startMsString == null ? null : Long.valueOf(startMsString);
+  }
+
+  static Long endMs(HttpServletRequest request) {
+    String endMsString = request.getParameter(END_MS_PARAM);
+    return endMsString == null ? null : Long.valueOf(endMsString);
+  }
+
+  static Pattern topic(HttpServletRequest request) {
+    String topicString = request.getParameter(TOPIC_PARAM);
+    return topicString != null ? Pattern.compile(topicString) : null;
+  }
+
+  static String resourceString(HttpServletRequest request) {
+    String resourceString = request.getParameter(RESOURCE_PARAM);
+    if (resourceString == null) {
+      resourceString = DEFAULT_PARTITION_LOAD_RESOURCE;
+    }
+    return resourceString;
+  }
+
+  static List<String> getGoals(HttpServletRequest request) {
+    boolean isKafkaAssignerMode = getMode(request);
+    List<String> goals;
+    String goalsString = request.getParameter(GOALS_PARAM);
+    // TODO: Handle urlencoded value (%2C instead of ,)
+    goals = goalsString == null ? new ArrayList<>() : Arrays.asList(goalsString.split(","));
+    goals.removeIf(String::isEmpty);
+
+    // If KafkaAssigner mode is enabled and goals are not specified, it overrides the goal list.
+    if (goals.isEmpty() && isKafkaAssignerMode) {
+      goals = Arrays.asList(KafkaAssignerEvenRackAwareGoal.class.getSimpleName(),
+                            KafkaAssignerDiskUsageDistributionGoal.class.getSimpleName());
+    }
+    return goals;
+  }
+
+  static int entries(HttpServletRequest request) {
+    String entriesString = request.getParameter(ENTRIES_PARAM);
+    return entriesString == null ? Integer.MAX_VALUE : Integer.parseInt(entriesString);
+  }
+
+  static int partitionLowerBoundary(HttpServletRequest request) {
+    String partitionString = request.getParameter(PARTITION_PARAM);
+    if (partitionString == null) {
+      return Integer.MIN_VALUE;
+    }
+
+    return !partitionString.contains("-") ? Integer.parseInt(partitionString)
+                                          : Integer.parseInt(partitionString.split("-")[0]);
```

Comment:
Should there be a strict handling here?
Example, `value=10-11-12-13` based on the code it will take the lower bound as 10 and the upper bound as 11.
I think we should report this as an error.